packages/vitest/src/runtime/execute.ts

```diff
@@ -32,12 +37,31 @@ let _viteNode: {
 export const moduleCache = new ModuleCacheMap()
 export const mockMap: MockMap = new Map()
 
-export async function startViteNode(ctx: ContextRPC) {
+export async function startViteNode(ctx: ContextRPC, options: ContextExecutorOptions) {
   if (_viteNode)
     return _viteNode
```

Comment:
Adding a note here while I remember this:

This also caches `rpc` and breaks it when running tests with following conditions:
- `--no-isolate`
- `--single-thread`
- Test files have multiple test environments in use, e.g. `node`, `jsdom`

Reply:
Another founding: Something makes `--no-isolate` stuck even without `--single-thread` and with a single test environment. To reproduce remove this line

https://github.com/sheremet-va/vitest/blob/1b0e6ad5d8cdcf404d8daba01524f1b26111b3ee/test/core/vitest.config.ts#L53-L55

... and run `test/core` tests with `pnpm run test --no-isolate`. Vitest gets stuck after few first initial tests. I have not yet found the root cause but I'm not sure whether it's related to the

Reply:
I think it's the same issue. One worker can be used for different tests, but vite node instance is the same.

Reply:
Maybe we should just not cache it 🤔

Reply:
Currently caching only affects `--no-isolate` 🤔 So we can either disable it or make tinypool smarter to not reuse the same worker for different workspaces. Because different tests will have the same config, but different rpc

Reply:
It should run in a separate process if test has different env, env options and or it's in another workspace

Reply:
> But I don't think it's this PR's issue, the same happens in main.

I agree. I can create a bug report for this later. Here's minimal reproduction:

<details>
  <summary>Minimal repro for main branch</summary>

```js
export default defineConfig({
  test: {
    isolate: false,
    singleThread: true,
  },
});
```

```js
/**
 * @vitest-environment node
 */
test("jsdom globals not found", async () => {
  globalThis.__leaking_from_node = "leaking";
  expect(globalThis.__leaking_from_jsdom).toBe(undefined);
});
```

```ts
/**
 * @vitest-environment jsdom
 */
test("Node globals not found", async () => {
  globalThis.__leaking_from_jsdom = "leaking";
  expect(globalThis.__leaking_from_node).toBe(undefined);
});
```

</details>

> So we can either disable it or make tinypool smarter to not reuse the same worker for different workspaces. 

I'm starting to think that we should add support for this in Tinypool. In future we may have other use cases where it would be ideal to use a single pool and be able to recycle a worker - either a new imperative method or through `pool.run` options.